src/schema/sources.ts

```diff
@@ -505,6 +515,31 @@ export const canAccessSource = async (
   );
 };
 
+export const canPostToSquad = async (
+  ctx: Context,
+  squad: SquadSource,
+): Promise<boolean> => {
+  if (squad.allowMemberPosting) {
+    return true;
+  }
+
+  const repo = ctx.getRepository(SourceMember);
+  const loggedUser = await repo.findOneByOrFail({
+    sourceId: squad.id,
+    userId: ctx.userId,
+  });
```

Comment:
If there is a way to reuse `loggedUser` to not do another query to the database I would be interested in example for that. 

I mean, there are already code paths that do this kind of thing but still curious.

Reply:
I think it's better to get the SourceMember as a parameter, just like the source. Probably before calling this function, we also fetch it somewhere, so it's better to reuse it as mentioned

Reply:
Solved with 4822c4664aa28c883b844ca397980dd1caf41493